javascript/stimulus_reflex.js

```diff
@@ -136,33 +128,30 @@ const register = (controller, options = {}) => {
             'selectors',
             'reflexId',
             'resolveLate',
-            'serializeForm'
+            'serializeForm',
+            'includeInnerHTML',
+            'includeTextContent'
           ].includes(key)
         ).length
       ) {
         const opts = args.shift()
         Object.keys(opts).forEach(o => (options[o] = opts[o]))
       }
-      const attrs = options['attrs'] || extractElementAttributes(reflexElement)
-      const reflexId = options['reflexId'] || uuidv4()
-      let selectors = options['selectors'] || getReflexRoots(reflexElement)
-      if (typeof selectors === 'string') selectors = [selectors]
-      const resolveLate = options['resolveLate'] || false
-      const dataset = extractElementDataset(reflexElement)
-      const xpathController = elementToXPath(controllerElement)
-      const xpathElement = elementToXPath(reflexElement)
+
```

Comment:
This is gonna blow up #505 but I'll live. ;)

Okay, I am highly in favour of this refactor but I would ask you to "go all of the way". In #505 I added two lines after 153 that declared `const`s for `reflexController` and `permanentAttributeName` so that everything in the `data` object could just be a dual key/value thing.

I would like to expand the method signature of `ReflexData#initialize` so that it accepts `target, args, url, tabId` as well as `reflexController, permanentAttributeName`. That way, there is no longer a need to create a `data` object. We can just call getters on the new `reflexData` object you returned, in the same way that you already access `reflexId` on line 147.

The current `data` object is only referenced on lines 173, 209 and 217.

Reply:
Right on! Back then I didn't wanna do too much but 👍🏻

Reply:
Okay, done! It does feel a bit of a data clump now to me, but it's been all contracted in a single location 👍

That said, immediately `dispatchLifecycleEvent` jumped at me, which could now potentially be passed an instance of `ReflexData`. I'll take a look at that next.

Reply:
This looks really good to me. Nicely done, and thanks for the iteration.

wrt `dispatchLifecycleEvent`: it's not going to be quite so simple, as it gets called in different contexts... also, ReflexData does not include `stage` or `payload`. So I'd say that you can table this, for now at least.

Reply:
Yeah, I noticed that later!